e_adaplio:

  # Topics
  pointCloudTopic: "/velodyne_points"               # Point cloud data
  imuTopic: "handsfree/imu"                         # IMU data
  odomTopic: "odometry/imu"                   # IMU pre-preintegration odometry, same frequency as IMU
  gpsTopic: "odometry/gpsz"                   # GPS odometry topic from navsat, see module_navsat.launch file

  # Frames
  lidarFrame: "base_link"
  baselinkFrame: "base_link"
  odometryFrame: "odom"
  mapFrame: "map"

  # GPS Settings
  useImuHeadingInitialization: false           # if using GPS data, set to "true"
  useGpsElevation: false                      # if GPS elevation is bad, set to "false"
  gpsCovThreshold: 2.0                        # m^2, threshold for using GPS data
  poseCovThreshold: 25.0                      # m^2, threshold for using GPS data
  
  # Export settings
  # savePCD: true                              # https://github.com/TixiaoShan/LIO-SAM/issues/3
  # savePCDDirectory: "/tmp/"        # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "mapOptimization" for implementation

  # # --------------------------------------------------------该部分是适配liosam数据集的--------------------------------------------------
  sensor: velodyne                            
  N_SCAN: 32                                 
  Horizon_SCAN: 1800                         
  downsampleRate: 1                           
  lidarMinRange: 1.0                         
  lidarMaxRange: 80.0                      
  # # --------------------------------------------------------该部分是适配liosam数据集的-------------------------------------------------

  imuAccNoise: 3.7686306102624571e-02
  imuGyrNoise: 2.3417543020438883e-03
  imuAccBiasN: 1.1416642385952368e-03
  imuGyrBiasN: 1.4428407712885209e-05
  imuGravity: 9.80511
  imuRPYWeight: 0.01

  # # --------------------------------------------------------该部分是适配liosam数据集的-----------------------------------------------------
  extrinsicTrans: [0.27255, -0.00053,0.17954]
  extrinsicRot: [1, 0, 0,
                  0, 1, 0,
                  0, 0, 1]
  extrinsicRPY: [1, 0, 0,
                  0, 1, 0,
                  0, 0, 1]              
  # # --------------------------------------------------------该部分是适配liosam数据集的-----------------------------------------------------


  # LOAM feature threshold
  edgeThreshold: 1.0
  surfThreshold: 0.1
  edgeFeatureMinValidNum: 10
  surfFeatureMinValidNum: 100

  # voxel filter paprams   分室内室外 
  # mappingCornerLeafSize: 0.2                   # default: 0.2 - outdoor, 0.1 - indoor
  # mappingSurfLeafSize: 0.4                      # default: 0.4 - outdoor, 0.2 - indoor

  # robot motion constraint (in case you are using a 2D robot)
  z_tollerance: 10                            # meters
  rotation_tollerance: 1000                     # radians

  # CPU Params
  numberOfCores: 4                              # number of cores for mapping optimization
  mappingProcessInterval: 0.15                  # seconds, regulate mapping frequency

  # Surrounding map
  surroundingkeyframeAddingDistThreshold: 20.0   # meters, regulate keyframe adding threshold，这里将 10.0 设定为 极丰富;打算在低速的时候设为10，高速的时候设为20
  surroundingkeyframeAddingAngleThreshold: 0.5  # radians, regulate keyframe adding threshold，这里将 0.5 设定为 极丰富 丰富度的值
  surroundingKeyframeDensity: 2.0               # meters, downsample surrounding keyframe poses   
  # surroundingKeyframeSearchRadius: 50.0         # meters, within n meters scan-to-map optimization (when loop closure disabled)

  # Loop closure
  loopClosureEnableFlag: true
  loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency
  surroundingKeyframeSize: 50                   # submap size (when loop closure enabled)
  historyKeyframeSearchRadius: 15.0             # meters, key frame that is within n meters from current pose will be considerd for loop closure
  historyKeyframeSearchTimeDiff: 30.0           # seconds, key frame that is n seconds older will be considered for loop closure
  historyKeyframeSearchNum: 25                  # number of hostory key frames will be fused into a submap for loop closure
  historyKeyframeFitnessScore: 0.3              # icp threshold, the smaller the better alignment

  # Visualization
  globalMapVisualizationSearchRadius: 1000.0    # meters, global map visualization radius
  globalMapVisualizationPoseDensity: 10.0       # meters, global map visualization keyframe density
  globalMapVisualizationLeafSize: 1.0           # meters, global map visualization cloud density